File: app/src/main/java/ru/laushkina/rates/util/RatesLog.java
package ru.laushkina.rates.util;

import androidx.annotation.NonNull;

import android.util.Log;

import ru.laushkina.rates.BuildConfig;

public class RatesLog {

    public static void d(@NonNull String tag, @NonNull String message) {
        if (BuildConfig.DEBUG) {
            Log.d(tag, message);
        }
    }

    public static void e(@NonNull String tag, @NonNull String message) {
        if (BuildConfig.DEBUG) {
            Log.e(tag, message);
        }
    }

    public static void e(@NonNull String tag, @NonNull String message, @NonNull Throwable throwable) {
        if (BuildConfig.DEBUG) {
            Log.e(tag, message, throwable);
        }
    }
}

File: app/src/main/java/ru/laushkina/rates/util/RatesLog.kt
package ru.laushkina.rates.util

import android.util.Log
import ru.laushkina.rates.BuildConfig

class RatesLog {
    companion object {
        fun d(tag: String, message: String) {
            if (BuildConfig.DEBUG) {
                Log.d(tag, message)
            }
        }

        fun e(tag: String, message: String) {
            if (BuildConfig.DEBUG) {
                Log.e(tag, message)
            }
        }

        fun e(tag: String, message: String, throwable: Throwable?) {
            if (BuildConfig.DEBUG) {
                Log.e(tag, message, throwable)
            }
        }
    }
}